src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunILLink.cs

```diff
@@ -311,6 +353,26 @@ private void RemoveRootDescriptor(XDocument project)
                 .First().Remove();
         }
 
+        private void EnableNonFrameworkTrimming(XDocument project)
+        {
+            var ns = project.Root.Name.Namespace;
+
+            var target = new XElement(ns + "Target",
+                                      new XAttribute("AfterTargets", "_SetILLinkDefaults"),
+                                      new XAttribute("Name", "_EnableNonFrameworkTrimming"));
```

Comment:
Why does the process for testing this involve a custom target running after a "private" sdk target. If this is a valid user scenario under test, then the project file needs to be expressible without this.

Reply:
It's not testing a user scenario. The user scenario we want to enable is the one that roots non-fx assemblies. Some of the other tests are checking that the linker is actually running by testing its ability to strip some specific code from our test project - which doesn't happen in the default mode, but is how we tell that the linker is running end-to-end.

Reply:
So the non default modes are unsupported? I still don't get why it's not possible to switch modes using "public" properties and no targets.

Reply:
Pretty much. Adding custom root assemblies or root descriptors will be supported, but we won't have a way to say "aggressive" and have that translate into a different set of specific illink options (at least in this preview). Someone who wants to do that will have to use the ILLink task directly or something similar.

Reply:
Can you add a comment to the test code here explaining why we're doing this?

Reply:
Sure!